### PR TITLE
8253342: Fix typos in String.transform

### PR DESCRIPTION
**Pull request by pavelrappo**
To confirm that this would not require a CSR I asked Joe Darcy. Joe replied that this would NOT require a CSR.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253342](https://bugs.openjdk.java.net/browse/JDK-8253342): Fix typos in String.transform

### Reviewers
 * [Jim Laskey](https://openjdk.java.net/census#jlaskey) (@JimLaskey - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/238/head:pull/238`
`$ git checkout pull/238`

### CODE REVIEW

File: src/java.base/share/classes/java/lang/String.java [hunk 1]
```diff
@@ -3180,12 +3180,12 @@ public String translateEscapes() {
      * string. The function should expect a single String argument
      * and produce an {@code R} result.
      * <p>
-     * Any exception thrown by {@code f()} will be propagated to the
+     * Any exception thrown by {@code f.apply()} will be propagated to the
      * caller.
      *
-     * @param f    functional interface to a apply
+     * @param f    a function to apply
```
Comment (shipilev) [inline src/java.base/share/classes/java/lang/String.java:3186]: Shouldn't this be "_the_ function to apply"? Grep for other instance of "function to apply" in javadocs :)

**Review by shipilev — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/java/lang/String.java (hunk 1 above)
Comment (pavelrappo) [reply src/java.base/share/classes/java/lang/String.java:3186]: I had grepped that before pushing and the results were inconclusive. For example, compare the below fragments from the same file:

```
    /**
...
     * @param mapper the mapping function to apply to each element
...
     */
    public static <T>
    Collector<T, ?, LongSummaryStatistics> summarizingLong(ToLongFunction<? super T> mapper)
```
```
    /**
...
     * @param mapper a mapping function to apply to each element
...
     */
    public static <T>
    Collector<T, ?, DoubleSummaryStatistics> summarizingDouble(ToDoubleFunction<? super T> mapper)
```

Comment (shipilev) [reply src/java.base/share/classes/java/lang/String.java:3186]: Hm! Better to defer to native speakers then...

**Review by shipilev — COMMENTED** [review]
Changing the review from "Request Changes" to just "Comment.

**Review by JimLaskey — APPROVED** [review]

**Review by JimLaskey — APPROVED** [review]

**Pull request closed without merging**